pandas/core/internals/managers.py

```diff
@@ -823,10 +831,11 @@ def as_array(
         else:
             arr = self._interleave(dtype=dtype, na_value=na_value)
 
+        if copy or na_value is not lib.no_default:
+            arr = arr.copy()
```

Comment:
When `interleave` is called, that will always be already a copy, so it would be nice to avoid a second copy there as well

Maybe the above if/else can be reorganized slightly, like:

```
if self._is_single_block or not self.is_mixed_type:
  if self.blocks[0].is_datetimetz:
    ...
  elif self.blocks[0].is_extension:
    ...
  else:
    ....
  if copy or na_value is not lib.no_default:
    arr = arr.copy()
else:
  arr = self._interleave(dtype=dtype, na_value=na_value)
```

Reply:
Good idea. Not sure if it's "bad style" but is another option simply to create a "should_copy" variable and update it for the interleave case? Might be a bit more explicit what we're thinking and helps avoid nested conditionals.

Reply:
Such a variable is also fine, whatever is easiest to implement / you prefer